addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanner.java

```diff
@@ -0,0 +1,126 @@
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import static net.htmlparser.jericho.HTMLElementName.LINK;
+import static net.htmlparser.jericho.HTMLElementName.SCRIPT;
+
+/** Detect missing attribute integrity in tag <script> */
+public class SubResourceIntegrityAttributeScanner extends PluginPassiveScanner {
+    /** Prefix for internationalized messages used by this rule */
+    private static final String MESSAGE_PREFIX = "pscanalpha.sri-integrity.";
+
+    // From
+    // https://w3c.github.io/webappsec-subresource-integrity/#verification-of-html-document-subresources
+    // To support integrity metadata for some of these elements, a new integrity attribute is added
+    // to
+    // the list of content attributes for the link and script elements.
+    // Note: A future revision of this specification is likely to include integrity support for all
+    // possible subresources, i.e., a, audio, embed, iframe, img, link, object, script, source,
+    // track,
+    // and video elements.
+    private static final List<String> SUPPORTED_ELEMENTS = Arrays.asList(SCRIPT, LINK);
+
+    private static final Map<String, String> CONTENT_ATTRIBUTES =
+            new HashMap<String, String>() {
+                {
+                    put(SCRIPT, "src");
+                    put(LINK, "href");
+                }
+            };
+
+    private PassiveScanThread parent;
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+        // do nothing
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        List<Element> sourceElements = source.getAllElements();
+        sourceElements.stream()
+                .filter(element -> SUPPORTED_ELEMENTS.contains(element.getName()))
+                .filter(unsafeSubResource(msg.getRequestHeader().getHostName()))
+                .forEach(
+                        element -> {
+                            Alert alert =
+                                    new Alert(
+                                            getPluginId(),
+                                            Alert.RISK_MEDIUM,
+                                            Alert.CONFIDENCE_HIGH,
+                                            getName());
+
+                            alert.setDetail(
+                                    getString("desc"),
+                                    msg.getRequestHeader().getURI().toString(),
+                                    "",
+                                    "",
+                                    "",
+                                    getString("soln"),
+                                    getString("refs"),
+                                    element.toString(),
+                                    287, // Improper Authentication
```

Comment:
Is this really the one you wanted? Doesn’t seem right to me.

Reply:
Not sure either. Could be CWE-16? Like that example: https://www.acunetix.com/vulnerabilities/web/subresource-integrity-sri-not-implemented/

Reply:
Ya that sounds good.

Reply:
Maybe also set WASC 14 then? http://projects.webappsec.org/w/page/13246975/Threat%20Classification%20Taxonomy%20Cross%20Reference%20View